File: little-bee/controller/router/alarm_level.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_ALARM_LEVEL = "/config/alarm-level"
const URL_ROUTER_ALARM_LEVEL_BY_ID = "/config/alarm-level/:alarm-level-id"
const URL_ROUTER_ALARM_LEVEL_QUERY = "/config/alarm-level/query"
const URL_ROUTER_ALARM_LEVEL_ALL = "/config/alarm-level/all"

func AlarmLevelRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_ALARM_LEVEL, config.CreateAlarmLevel)
	r.GET(URL_ROUTER_ALARM_LEVEL_BY_ID, config.RetrieveAlarmLevel)
	r.PUT(URL_ROUTER_ALARM_LEVEL, config.UpdateAlarmLevel)
	r.DELETE(URL_ROUTER_ALARM_LEVEL_BY_ID, config.DeleteAlarmLevel)
	r.PUT(URL_ROUTER_ALARM_LEVEL_BY_ID, config.DeleteAlarmLevelWithFlag)
	r.POST(URL_ROUTER_ALARM_LEVEL_QUERY, config.QueryAlarmLevel)
	r.GET(URL_ROUTER_ALARM_LEVEL_ALL, config.GetAllAlarmLevels)
	return r
}

File: frontend/console/src/api/alarm-level.js
import Axios from "@/http";

export function createAlarmLevel(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/alarm-level', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveAlarmLevel(alarmLevelId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/alarm-level/${ alarmLevelId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateAlarmLevel(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/alarm-level', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteAlarmLevel(alarmLevelId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/alarm-level/${ alarmLevelId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteAlarmLevelWithFlag(alarmLevelId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/alarm-level/${ alarmLevelId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllAlarmLevels() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/alarm-level/all').then(res => {
            resolve(res)
        })
    })
}

export function queryAlarmLevel (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/alarm-level/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}